src/aws/nodemanager/nodemanager.py

```diff
@@ -71,18 +99,34 @@ def generate_heartbeat(self, notify=True):
 
     def process_heartbeat(self, hb, source) -> Packet:
         hb['source'] = source  # Set the source IP of the heartbeat (i.e., the worker).
-
+        self.notify(hb)  # Forward the heartbeat to the monitor for metrics.
+        if len(self._tasks)>0:  #If there are tasks in the taskpool send a new command to the worker
+            packet=CommandPacket(command="task")
+            current_task=self._tasks.pop(0)
+            packet["task_data"]= current_task.get_task_data()
+            current_task.state=TaskState.RUNNING
+            self._tasks_running.append(current_task)
+            return packet
+        else:
+            return hb #In case there are no more commands send hb           
         # TODO: process the heartbeat and take actions which task to send where @Karan.
         # In the heartbeat you could indicate how far the job is, if it is done, etc.
         # Based on this task 'state', you could assign new tasks, check if a new task should be
         # assigned to the worker, if a task should be stolen from the worker, etc.
 
-        # TODO: replace the hb below with command packet is work stealing, or new work is assigned.
-        return hb  # This value is returned to the worker client.
+        # This value is returned to the worker client.
+
+    def process_command(self, command: CommandPacket):
+        if command["command"]=="done":
+            
+            if len(self._tasks)>0:  #If there are tasks in the taskpool send a new command to the worker
```

Comment:
The same applies here as above. Now a new task is sent if the last tasks finished. However, workers can take multiple tasks and process them in FIFO order. We need to think about when and which tasks to send to which worker.